src/pattern/labels-to-string.ts

```diff
@@ -21,20 +21,31 @@ import type { CypherEnvironment } from "../Environment";
 import { LabelExpr } from "../expressions/labels/label-expressions";
 import { addLabelToken } from "../utils/add-label-token";
 import { asArray } from "../utils/as-array";
-import { escapeLabel } from "../utils/escape";
+import { escapeLabel, escapeType } from "../utils/escape";
 
 export function labelsToString(labels: string | string[] | LabelExpr, env: CypherEnvironment): string {
     if (labels instanceof LabelExpr) {
-        const labelsStr = labels.getCypher(env);
-        if (!labelsStr) {
-            return "";
-        }
         return addLabelToken(env.config.labelOperator, labels.getCypher(env));
     } else {
-        const escapedLabels = asArray(labels).map(escapeLabel);
-        if (escapedLabels.length === 0) {
-            return "";
+        let escapedLabels = asArray(labels);
+
+        if (!env.config.unsafeEscapeOptions.disableLabelEscaping) {
```

Comment:
I would avoid the `not` and `disable` in the same condition, I would do an early return if it's disabled instead

Reply:
early return here means duplicating the call to `addLabelToken(env.config.labelOperator, ...escapedLabels)`, which I find even more confusing

Maybe a variable `shouldEscape`?